src/domains/services/config.service.ts

```diff
@@ -0,0 +1,123 @@
+import path from 'path';
+import os from 'os';
+import { promises as fs } from 'fs';
+
+const CONFIG_DIR = path.join(os.homedir(), '.asyncapi');
+const CONFIG_FILE = path.join(CONFIG_DIR, 'config.json');
+
+export interface AuthEntry {
+  pattern: string;
+  token: string;
+  authType?: string;
+  headers?: Record<string, string>;
+}
+
+export interface AuthResult {
+  token: string;
+  authType: string;
+  headers: Record<string, string>;
+}
+
+interface Config {
+  auth?: AuthEntry[];
+}
+
+export class ConfigService {
+  /**
+   * Load config file (~/.asyncapi/config.json)
+   */
+  static async loadConfig(): Promise<Config> {
+    try {
+      const content = await fs.readFile(CONFIG_FILE, 'utf8');
+      return JSON.parse(content) as Config;
+    } catch (err: any) {
+      if (err.code === 'ENOENT') {
+        return {}; // no config yet
+      }
+      throw new Error(`Error reading config file: ${err.message}`);
+    }
+  }
+
+  /**
+   * Save config back to file
+   */
+  static async saveConfig(config: Config): Promise<void> {
+    await fs.mkdir(CONFIG_DIR, { recursive: true });
+    await fs.writeFile(CONFIG_FILE, JSON.stringify(config, null, 2), 'utf8');
+  }
+
+  /**
+ * Add or update an auth entry by merging with existing entry if present
+ */
+  static async addAuthEntry(entry: AuthEntry): Promise<void> {
+    const config = await this.loadConfig();
+    if (!config.auth) {
+      config.auth = [];
+    }
+
+    // Find if an entry with the same pattern already exists
+    const index = config.auth.findIndex(e => e.pattern === entry.pattern);
+
+    if (index !== -1) {
+    // If found, merge existing entry with new entry
+      // eslint-disable-next-line security/detect-object-injection
```

Comment:
Is there a good reason to disable security/detect-object-injection? We're dealing with auth and tokens, shouldn't we be careful here?